server/command/command.go

```diff
@@ -0,0 +1,102 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+
+	"github.com/mattermost/mattermost-plugin-cloudapps/server/configurator"
+	"github.com/mattermost/mattermost-plugin-cloudapps/server/constants"
+	"github.com/mattermost/mattermost-plugin-cloudapps/server/utils/md"
+)
+
+type Mattermost interface {
+	RegisterCommand(*model.Command) error
+}
+
+type Command interface {
+	Init(*configurator.BuildConfig) error
+	Handle(pluginContext *plugin.Context, commandArgs *model.CommandArgs) (out md.MD, err error)
+}
+
+type command struct {
+	Mattermost
+	configurator.Configurator
+}
+
+var _ Command = (*command)(nil)
+
+func NewCommand(configurator configurator.Configurator, mm Mattermost) Command {
+	return &command{
+		Mattermost:   mm,
+		Configurator: configurator,
+	}
+}
+
+// Register should be called by the plugin to register all necessary commands
+func (c *command) Init(buildConfig *configurator.BuildConfig) error {
```

Comment:
Since apps will be able to register commands, we should put that logic somewhere. Not sure if this package is the right place or not.

Reply:
A future PR, a registration API for all things. Probably best elsewhere (in cloudapps[/commands]?)